vendor.mod

```diff
@@ -0,0 +1,198 @@
+)
+
+replace (
+	cloud.google.com/go => cloud.google.com/go v0.44.3
+	cloud.google.com/go/logging => cloud.google.com/go/logging v1.0.1-0.20190813144457-ceeb313ad77b
+	github.com/armon/go-metrics => github.com/armon/go-metrics v0.0.0-20150106224455-eb0af217e5e9
+	github.com/armon/go-radix => github.com/armon/go-radix v0.0.0-20150105235045-e39d623f12e8
+	github.com/containerd/containerd => github.com/containerd/containerd v1.5.8
+	github.com/coreos/go-semver => github.com/coreos/go-semver v0.2.0
+	github.com/coreos/go-systemd => github.com/coreos/go-systemd v0.0.0-20180511133405-39ca1b05acc7
+	github.com/coreos/pkg => github.com/coreos/pkg v0.0.0-20180108230652-97fdf19511ea
+	github.com/docker/distribution => github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible
+	github.com/gogo/googleapis => github.com/gogo/googleapis v1.3.2
+	github.com/golang/protobuf => github.com/golang/protobuf v1.3.5
+	github.com/google/go-cmp => github.com/google/go-cmp v0.2.0
+	github.com/google/uuid => github.com/google/uuid v1.1.1
+	github.com/hashicorp/go-immutable-radix => github.com/tonistiigi/go-immutable-radix v0.0.0-20170803185627-826af9ccf0fe
+	github.com/hashicorp/go-msgpack => github.com/hashicorp/go-msgpack v0.0.0-20140221154404-71c2886f5a67
+	github.com/hashicorp/go-multierror => github.com/hashicorp/go-multierror v1.0.0
+	github.com/hashicorp/serf => github.com/hashicorp/serf v0.7.1-0.20160317193612-598c54895cc5
+	github.com/hpcloud/tail => github.com/hpcloud/tail v1.0.0
+	github.com/klauspost/compress => github.com/klauspost/compress v1.11.13
+	github.com/matttproud/golang_protobuf_extensions => github.com/matttproud/golang_protobuf_extensions v1.0.1
+	github.com/onsi/ginkgo => github.com/onsi/ginkgo v1.8.0
+	github.com/onsi/gomega => github.com/onsi/gomega v1.5.0
+	github.com/pelletier/go-toml => github.com/pelletier/go-toml v1.8.1
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.6.0
+	github.com/prometheus/common => github.com/prometheus/common v0.9.1
+	github.com/prometheus/procfs => github.com/prometheus/procfs v0.0.11
+	github.com/stretchr/testify => github.com/stretchr/testify v1.3.0
+	github.com/vishvananda/netlink => github.com/vishvananda/netlink v1.1.0
+	go.opencensus.io => go.opencensus.io v0.22.3
+	golang.org/x/crypto => golang.org/x/crypto v0.0.0-20210322153248-0c34fe9e7dc2
+	golang.org/x/net => golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
+	golang.org/x/oauth2 => golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
+	golang.org/x/text => golang.org/x/text v0.3.3
+	golang.org/x/time => golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e
+	google.golang.org/api => google.golang.org/api v0.8.0
+	google.golang.org/genproto => google.golang.org/genproto v0.0.0-20200227132054-3f1135a288c9
+	google.golang.org/grpc => google.golang.org/grpc v1.27.1
+	gopkg.in/fsnotify.v1 => gopkg.in/fsnotify.v1 v1.4.7
+        archive/tar => ./internal/archive/tar
```

Comment:
So this actually isn't supported (see https://github.com/golang/go/issues/35283). We need to find a better way to do this, perhaps for now there could be `cp -a internal/archive/tar` inside of `go mod vendor` wrapper.

Reply:
If we keep the files inside `vendor/`, does `go mod vendor` delete them, or does it leave them alone?

also wondering if a symlink would work (although Windows may not be happy with that, I recall there were some issues with Windows and symlinks, but don't recall the details)

Reply:
> If we keep the files inside `vendor/`, does `go mod vendor` delete them, or does it leave them alone?

It deletes them.

> also wondering if a symlink would work (although Windows may not be happy with that, I recall there were some issues with Windows and symlinks, but don't recall the details)

The issue with symlinks on Windows is pretty major IMO, a Windows user basically cannot clone a repo with symlinks.

Reply:
Maybe I asked before, but why can't we move `vendor/archive/tar` to an external module such as `github.com/moby/sys/archive/tar` and just apply `s|archive/tar|github.com/moby/sys/archive/tar|g` to the moby/moby source?

Reply:
I think it might be worse trying as a follow-up, but here the goal is to make the least number of changes.

Reply:
@AkihiroSuda In that case, we could as well put it in `internal/` in this repository. But I think the reason we didn't "fork" it (but override) is that it may be used indirectly(?). Are we sure that the archive/tar code is never called indirectly (and we would reintroduce the vulnerability?)